pkg/gce-pd-csi-driver/node.go

```diff
@@ -302,3 +318,28 @@ func (ns *GCENodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGe
 func (ns *GCENodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
 	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("NodeExpandVolume is not yet implemented"))
 }
+
+func (ns *GCENodeServer) GetVolumeLimits() (int64, error) {
+	// Default value 16 for shared-CPU VM
+	volumeLimits := VolumeLimit16
+
+	// Machine-type format: n1-type-NCPU
+	machineType := ns.MetadataService.GetMachineType()
+	if strings.HasPrefix(machineType, "n1-") {
```

Comment:
custom machine types are also supported now and have the [syntax](https://cloud.google.com/sdk/gcloud/reference/container/clusters/create):

custom-cpus-memory

where memory is in MiB.

Reply:
Updated the rules

Reply:
I should have checked the original website, instead of the translated one..... It's official but obsolete